src/battle/star/focus_789E60.c

```diff
@@ -1,25 +1,60 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "battle/star/focus_789E60", func_802A1000_789E60);
-
-INCLUDE_ASM(s32, "battle/star/focus_789E60", func_802A10AC_789F0C);
-
-INCLUDE_ASM(s32, "battle/star/focus_789E60", func_802A116C_789FCC);
+#define NAMESPACE battle_star_focus
 
-INCLUDE_ASM(s32, "battle/star/focus_789E60", func_802A1218_78A078);
+#include "common/UnkStarFuncs.inc.c"
 
-#define NAMESPACE battle_star_focus
 #include "common/FadeBackgroundToBlack.inc.c"
 
-INCLUDE_ASM(s32, "battle/star/focus_789E60", func_802A137C_78A1DC);
+#include "common/UnkBackgroundFunc2.inc.c"
 
-INCLUDE_ASM(s32, "battle/star/focus_789E60", func_802A1414_78A274);
+#include "common/UnkBackgroundFunc.inc.c"
 
 INCLUDE_ASM(s32, "battle/star/focus_789E60", func_802A1494_78A2F4);
 
-// Duplicate of func_802A14E8_7905B8
-INCLUDE_ASM(s32, "battle/star/focus_789E60", func_802A14E8_78A348);
+#include "common/SetNpcCollision32.inc.c"
+
+ApiStatus func_802A1518_78A378(ScriptInstance* script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+    PlayerData* playerData2 = &gPlayerData;
+    s32 deepFocusSP;
+    s32 superFocusSP;
+
+    script->varTable[0] = 0;
+
+    if (playerData->specialBarsFilled >= playerData->maxStarPower * 256) {
+        script->varTable[0] = 1;
+        return ApiStatus_DONE2;
+    }
+
+    deepFocusSP = is_ability_active(Ability_DEEP_FOCUS) * 64;
+    superFocusSP = is_ability_active(Ability_SUPER_FOCUS) * 128;
+
+    add_SP(deepFocusSP + superFocusSP + 128);
+
+    if (playerData2->specialBarsFilled == playerData2->maxStarPower * 256) {
```

Comment:
Macro this condition and the other uses of it
Something like
```c
if ({
    PlayerData* playerData2 = &gPlayerData;
    playerData2->specialBarsFilled == playerData2->maxStarPower * 256;
}) {
    // ....
}
```

Reply:
Is it just the condition or the whole block? Or just like one of the fields? it's super unclear and I think it might be clearer if we wait and reconcile many examples where playerdata2 is needed

Reply:
just the condition, I'm fairly sure